test/generate.test.ts

```diff
@@ -184,22 +185,44 @@ describe('generate', function () {
       };
 
       const answerSet = generate(specQ, schema, stats);
+      assert.equal(answerSet.length, 3);
+      assert.equal((answerSet[0].getEncodingQueryByIndex(0).scale as ScaleQuery).type, ScaleType.TIME);
+      assert.equal((answerSet[1].getEncodingQueryByIndex(0).scale as ScaleQuery).type, ScaleType.UTC);
+      assert.equal((answerSet[2].getEncodingQueryByIndex(0).scale as ScaleQuery).type, undefined);
+    });
+
+    it('should enumerate correct scaleType for temporal field with timeunit', () => { // working on rn
+      const specQ = {
+        mark: Mark.POINT,
+        encodings: [
+          {
+            channel: Channel.X,
+            scale: {type: {values: [ScaleType.TIME, ScaleType.UTC, ScaleType.ORDINAL, undefined]}},
+            field: 'T',
+            type: Type.TEMPORAL,
+            timeUnit: TimeUnit.MINUTES
+          }
+        ]
+      };
+
+      const answerSet = generate(specQ, schema, stats);
       assert.equal(answerSet.length, 4);
       assert.equal((answerSet[0].getEncodingQueryByIndex(0).scale as ScaleQuery).type, ScaleType.TIME);
       assert.equal((answerSet[1].getEncodingQueryByIndex(0).scale as ScaleQuery).type, ScaleType.UTC);
       assert.equal((answerSet[2].getEncodingQueryByIndex(0).scale as ScaleQuery).type, ScaleType.ORDINAL);
       assert.equal((answerSet[3].getEncodingQueryByIndex(0).scale as ScaleQuery).type, undefined);
     });
 
-    it('should enumerate correct scaletype for ordinal field', () => {
+    it('should enumerate correct scaleType for ordinal field with timeunit', () => {
```

Comment:
Similar to N, you shouldn't use O with timeUnit (at least that's how it works in Vega-Lite for now).  We might revise this in the future for O.   (But definitely, not for N)